src/range.rs

```diff
@@ -1,98 +1,240 @@
 use core::{fmt, ops::Range};
 
-use crate::{PhysAddr, VirtAddr};
+use crate::{MemoryAddr, PhysAddr, VirtAddr};
 
 macro_rules! usize {
     ($addr:expr) => {
-        ($addr).as_usize()
+        Into::<usize>::into($addr)
     };
 }
 
-macro_rules! def_range {
-    ($name:ident, $addr_type:ty) => {
-        #[derive(Clone, Copy, Default, PartialEq, Eq)]
-        #[doc = concat!("A range of [`", stringify!($addr_type), "`].\n\n")]
-        #[doc = "The range is inclusive on the start and exclusive on the end."]
-        #[doc = "It is empty if `start >= end`."]
-        pub struct $name {
-            /// The lower bound of the range (inclusive).
-            pub start: $addr_type,
-            /// The upper bound of the range (exclusive).
-            pub end: $addr_type,
-        }
+/// A range of a given memory address type `A`.
+///
+/// The range is inclusive on the start and exclusive on the end.
+/// It is empty if `start >= end`.
+///
+/// ## Example
+///
+/// ```
+/// use memory_addr::VirtAddrRange;
```

Comment:
Is it still appropriate to use VirtAddrRange here?

Reply:
fixed.